ts/Series/SeriesOnPointComposition.ts

```diff
@@ -0,0 +1,517 @@
+                defined(connectedPoint.plotY)
+            ) {
+                const xFrom = defined(position.x) ?
+                        position.x :
+                        connectedPoint.plotX,
+                    yFrom = defined(position.y) ?
+                        position.y :
+                        connectedPoint.plotY,
+                    xTo = xFrom + (position.offsetX || 0),
+                    yTo = yFrom + (position.offsetY || 0),
+                    width = (connectorOpts && connectorOpts.width) || 1,
+                    color =
+                        (connectorOpts && (connectorOpts as any).color) ||
+                        this.color,
+                    dashStyle =
+                        connectorOpts && (connectorOpts as any).dashStyle;
+
+                let attribs: SVGAttributes = {
+                    d: SVGRenderer.prototype.crispLine([
+                        ['M', xFrom, yFrom],
+                        ['L', xTo, yTo]
+                    ], width, 'ceil')
+                };
+
+                attribs['stroke-width'] = width;
+
+                if (!chart.styledMode) {
+                    attribs.stroke = color;
+                    attribs.dashstyle = dashStyle;
+                }
+
+                return attribs;
+            }
+        }
+    }
+
+    /**
+     * Draw connector line that connects series and initial point's position.
+     * @private
+     */
+    function seriesDrawConnector(this: Series): void {
+        if (this.onPoint) {
+            if (!this.onPoint.connector) {
+                (this.onPoint.connector as any) = this.chart.renderer.path()
+                    .addClass('highcharts-connector-seriespoint')
+                    .attr({
+                        zIndex: -1
+                    })
+                    .add(this.markerGroup);
+            }
+
+            const attribs = this.getConnectorAttributes();
+            if (this.onPoint.connector) {
+                (this.onPoint.connector as any)['attr'](attribs);
+            }
+        }
+    }
+
+    function seriesDrawPoints(this: Series): void {
+        (drawPointsFunctions as any)[this.type + 'DrawPoints'].call(this);
+
+        this.seriesDrawConnector();
+    }
+
+    /* *
+     *
+     *  Classes
+     *
+     * */
+
+    /**
+     * @private
+     */
+    export class Additions {
+
+        /* *
+         *
+         *  Constructors
+         *
+         * */
+
+        /**
+         * @private
+         */
+        public constructor(series: SeriesComposition) {
+            this.series = series;
+            this.options = series.options.onPoint;
+        }
+
+        /* *
+         *
+         *  Properties
+         *
+         * */
+
+        public series: SeriesComposition;
+
+        public connector?: SVGElement = void 0 as any;
+
+        public options?: OnPoint;
+    }
+
+}
+
+/* *
+ *
+ *  Default Export
+ *
+ * */
+
+export default SeriesOnPointComposition;
+
+/* *
+ *
+ *  API Options
```

Comment:
All API options should be available only for the selected series that supports this feature (so pie and sunburst for now). Right now the options are visible for every series type like area or any indicator in Stock.